core/src/main/java/org/infinispan/expiration/impl/ExpirationManagerImpl.java

```diff
@@ -212,6 +210,11 @@ private void deleteFromStoresAndNotify(K key, V value, Metadata metadata, Privat
             cacheNotifier.notifyCacheEntryExpired(key, value, metadata, null)));
    }
 
+   private void notify(K key, V value, Metadata metadata, PrivateMetadata privateMetadata) {
+      listeners.forEach(l -> l.expired(key, value, metadata, privateMetadata)); //for internal use, assume non-blocking
```

Comment:
The comment is the exact same one as the method above. I personally like the comment as the method doesn't even allow for the code to be blocking since it has the note about being internal only.